src/octoprint/static/js/app/viewmodels/files.js

```diff
@@ -423,16 +424,74 @@ $(function() {
                     }
                 }
                 output += gettext("Estimated Print Time") + ": " + formatDuration(data["gcodeAnalysis"]["estimatedPrintTime"]) + "<br>";
-		if(self.loginState.isUser()&&data["gcodeAnalysis"]["warning"])
+		if(self.loginState.isUser())
 		{
-		    var warning = "<p>" + gettext("Object exceeds printing area") + "</p>";
-                    warning += pnotifyAdditionalInfo("<pre>Object or objects in file '"+data["name"]+"' exceed/s the printing area</pre>");
-                    new PNotify({
-                        title: "Object dimensions",
-                        text: warning,
-                        type: "warning",
-                        hide: false
-                    });
+                    var profileItems=self.printerProfiles.profiles.items();
+                    var count=0;
+                    while(count<profileItems.length && !profileItems[count].current)
+                    {
+                        count++;
+                    }
+                    if(count<profileItems.length)
+                    {
+                        var warn=false;
+                        var volumeInfo=profileItems[count].volume;
+                        var printingArea=data["gcodeAnalysis"]["printingArea"];
+                        if(volumeInfo.origin=="lowerleft")
+                        {
+                            if(printingArea["maxX"]>volumeInfo.width || printingArea["maxY"]>volumeInfo.depth || printingArea["maxZ"]>volumeInfo.height)
+                            {
+                                warn=true;
+                            }
+                        }
+                        else //origin is centre
+                        {
+                            if( printingArea["maxX"]>(volumeInfo.width/2) || printingArea["maxY"]>(volumeInfo.depth/2) || printingArea["maxZ"]>volumeInfo.height || Math.abs(printingArea["minX"])>(volumeInfo.width/2) || Math.abs(printingArea["minY"])>(volumeInfo.depth/2))
+                            {
+                                warn=true;
+                            }
+                        }
+                    }
+                    if(warn)
```

Comment:
I agree this way is much better than the improvement I applied to the original code.